recirq/engine_utils_test.py

```diff
@@ -108,3 +115,36 @@ def test_sampler_by_name():
     assert isinstance(recirq.get_sampler_by_name('Syc23-zeros',
                                                  gateset='sqrt-iswap'),
                       recirq.ZerosSampler)
+
+
+@patch('cirq.google.engine.engine_client.quantum.QuantumEngineServiceClient')
+@patch('recirq.engine_utils._get_current_time')
+@patch('cirq.google.engine.EngineProcessor.get_schedule')
+def test_get_available_processors_open_swim_1(schedule_mock, time_mock, engine_mock):
+    os.environ['GOOGLE_CLOUD_PROJECT'] = 'some_project'
+    schedule_mock.return_value = [
+        qtypes.QuantumTimeSlot(
+            processor_name='Sycamore23',
+            start_time=Timestamp(seconds=100),
+            end_time=Timestamp(seconds=500),
+            slot_type=enums.QuantumTimeSlot.TimeSlotType.OPEN_SWIM)
+    ]
+    time_mock.return_value = datetime.fromtimestamp(300)
+    assert 'Sycamore23' in recirq.get_available_processors(['Sycamore23'])
+
+
+@patch('cirq.google.engine.engine_client.quantum.QuantumEngineServiceClient')
+@patch('recirq.engine_utils._get_current_time')
+@patch('cirq.google.engine.EngineProcessor.get_schedule')
+def test_get_available_processors_open_swim_2(schedule_mock, time_mock, engine_mock):
+    os.environ['GOOGLE_CLOUD_PROJECT'] = 'some_project'
+    schedule_mock.return_value = [
+        qtypes.QuantumTimeSlot(
+            processor_name='Sycamore23',
+            start_time=Timestamp(seconds=300),
+            end_time=Timestamp(seconds=500),
+            slot_type=enums.QuantumTimeSlot.TimeSlotType.OPEN_SWIM)
+    ]
+    time_mock.return_value = datetime.fromtimestamp(700)
+    assert recirq.get_available_processors(['Sycamore23']) == []
```

Comment:
Let's add tests for reservations with/without the project id.

Reply:
Done